lua/orgmode/org/mappings.lua

```diff
@@ -462,6 +462,147 @@ function OrgMappings:show_help()
   return Help.show()
 end
 
+function OrgMappings:edit_src_block()
+  local start_bufnr = vim.api.nvim_get_current_buf()
+  local file = Files.get_current_file()
+  local pos = vim.api.nvim_win_get_cursor(0)
+
+  local get_blocks = function()
+    -- Source blocks might not have content, which is fine
+    local content_query = '(block (name) @name (parameters) @parameters (contents) @contents (#eq? @name "SRC"))'
+    local no_content_query = '(block (name) @name (parameters) @parameters (#eq? @name "SRC"))'
+
+    local blocks = file:get_ts_matches(content_query)
+    if #blocks == 0 then
+      blocks = file:get_ts_matches(no_content_query)
+
+      if #blocks == 0 then
+        return nil
+      end
+    end
+
+    return blocks
+  end
+
+  local get_src_block_info = function(ts_blocks, pos_info)
+    local target_block
+
+    for _, block in ipairs(ts_blocks) do
+      if ts_utils.is_in_node_range(block.name.node:parent(), pos_info[1], pos_info[2]) then
+        target_block = block
+        break
+      end
+    end
+
+    if not target_block then
+      return nil, nil, nil
+    end
+
+    return target_block.name, target_block.contents, target_block.parameters
+  end
+
+  local blocks = get_blocks()
+  if not blocks then
+    utils.echo_error('Unable to edit source block')
+    return
+  end
+
+  -- The block might not have content yet, which is fine
+  local name, contents, parameters = get_src_block_info(blocks, pos)
+  if not name or not parameters then
+    utils.echo_error('No valid source block found')
+    return
+  end
+
+  local content = contents and contents.text_list or {}
+  local ft = parameters.text
+  local src_start_line = name.node:parent():range()
+
+  if type(config.org_edit_src_block.open) ~= 'function' then
+    utils.echo_error("Invalid 'org_edit_src_block' setting, 'open' is not a function")
+    return
+  end
+
+  local bufnr = config.org_edit_src_block.open({
+    bufnr = start_bufnr,
+    cursor = pos,
+    file = file,
+    ft = ft,
+    nodes = {
+      name = name,
+      contents = contents,
+    },
+  })
+
+  if not bufnr or not vim.api.nvim_buf_is_valid(bufnr) then
+    utils.echo_info('Cancel editing source block')
+    return
+  end
+
+  local funcname = string.format('_org_edit_src_block_%s', bufnr)
+  _G[funcname] = function(wipe)
```

Comment:
I don't think it's needed to create a global function for each buffer. It should work the same for all edits. Open in tmp buffer, save, return to previous buffer and update the block. 
For the state that you need, just put it in the buffer variable and use it when needed. You can save the file+node range in the buffer variable so you know what to update after.
Functionality should be really similar to the "capture" functionality. See how capture functionality is done https://github.com/kristijanhusak/orgmode.nvim/blob/master/lua/orgmode/capture/init.lua#L39

Reply:
Makes sense, I've implemented it in a similar fashion and changed the callback to happen on `BufWipeout` instead of `BufWritePost`. The latter feels a bit more "responsive" but granted it can be a bit overkill (and useless if the user decides to specify something like `tabnew` for editing the temporary buffer).